# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerRegisterJob.h
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/serviceworkers/ServiceWorkerRegisterJob.h
--- a/dom/serviceworkers/ServiceWorkerRegisterJob.h	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/serviceworkers/ServiceWorkerRegisterJob.h	Thu Oct 24 03:02:42 2024 +0000
@@ -16,9 +16,11 @@
 // spec algorithms.
 class ServiceWorkerRegisterJob final : public ServiceWorkerUpdateJob {
  public:
-  ServiceWorkerRegisterJob(nsIPrincipal* aPrincipal, const nsACString& aScope,
-                           const nsACString& aScriptSpec,
-                           ServiceWorkerUpdateViaCache aUpdateViaCache);
+  ServiceWorkerRegisterJob(
+      nsIPrincipal* aPrincipal, const nsACString& aScope,
+      const nsACString& aScriptSpec,
+      ServiceWorkerUpdateViaCache aUpdateViaCache,
+      const ServiceWorkerLifetimeExtension& aLifetimeExtension);
 
  private:
   // Implement the Register algorithm steps and then call the parent class